btc_light_client.go

```diff
@@ -46,25 +47,51 @@ func (lc *BTCLightClient) FindPreviousCheckpoint() (blockchain.HeaderCtx, error)
 	return nil, nil
 }
 
-func (lc *BTCLightClient) AddHeader(height int64, header wire.BlockHeader) error {
-	return lc.btcStore.AddHeader(height, header)
+func (lc *BTCLightClient) SetHeader(height int64, header wire.BlockHeader) error {
+	return lc.btcStore.SetHeader(height, header)
 }
 
 // We assume we always insert valid header. Acctually, Cosmos can revert a state
 // when module return error so this assumtion is reasonable
 func (lc *BTCLightClient) InsertHeaders(headers []wire.BlockHeader) error {
 	latestHeight := lc.btcStore.LatestHeight()
-	for _, header := range headers {
+
+	return lc.insertHeadersWithPosition(uint64(latestHeight), headers)
+}
+
+func (lc *BTCLightClient) insertHeadersWithPosition(height uint64, headers []wire.BlockHeader) error {
+	insertHeight := height + 1
```

Comment:
We should rename to insertHeaderStartAtHeigh(height,  headers). This is more meanful.